src/app-layout/visual-refresh-toolbar/toolbar/trigger-button/index.tsx

```diff
@@ -44,11 +75,137 @@ function TriggerButton(
     badge,
     selected = false,
     highContrastHeader,
+    hasTooltip = false,
+    tooltipText,
+    hasOpenDrawer = false,
+    isMobile = false,
+    isForPreviousActiveDrawer = false,
+    isForSplitPanel = false,
   }: TriggerButtonProps,
   ref: React.Ref<ButtonProps.Ref>
 ) {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const tooltipValue = tooltipText ? tooltipText : ariaLabel ? ariaLabel : '';
+  const [showTooltip, setShowTooltip] = useState<boolean>(false);
+  const [suppressTooltip, setSupressTooltip] = useState<boolean>(false);
+
+  const handleTriggerClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.stopPropagation(); // Stop the event from propagating to the badge icon
+    setShowTooltip(false);
+    setSupressTooltip(true);
+    onClick(event);
+  };
+
+  const handleBlur = (keepSupressed = false) => {
+    setSupressTooltip(keepSupressed);
+    setShowTooltip(false);
+  };
+
+  const handlePointerEnter = () => {
+    setSupressTooltip(false);
+    setShowTooltip(true);
+  };
+
+  /**
+   * Takes the drawer being closed and the data-shift-focus value from a close button on that drawer that persists
+   * on the event relatedTarget to determine not to show the tooltip
+   * @param event
+   */
+  const handleOnFocus = useCallback(
+    (event: FocusEvent) => {
+      let shouldShowTooltip = false;
+      const eventWithRelatedTarget = event as any;
+      const relatedTarget = eventWithRelatedTarget?.relatedTarget;
+      const isFromAnotherTrigger = relatedTarget?.dataset?.shiftFocus === 'awsui-layout-drawer-trigger';
+      if (
+        isForSplitPanel || //for tab/key navigation to button from breadcrume
```

Comment:
Nitpick: typo in comment (`breadcrume`)

Reply:
Is it ever possible that `isForSplitPanel` and `isForPreviousActiveDrawer` are true at the same time? If not, there is no need to check for `isForSplitPanel` in this condition.

Reply:
yes. when the split panel is open that both would be true